### PR TITLE
Add date formats options for tabular date type

### PR DESCRIPTION
**Pull request by bewakes**

### CODE REVIEW

File: src/components/other/TabularSheet/EditField/index.js [hunk 1]
```diff
@@ -52,6 +76,7 @@ class EditFieldModal extends React.PureComponent {
             options: { $auto: {
                 separator: { $setDefault: 'none' },
                 geoType: { $setDefault: 'name' },
+                dateFormat: { $setDefault: undefined },
```
Comment (tnagorra) [inline src/components/other/TabularSheet/EditField/index.js]: Setting default to undefined is un-necessary

File: src/components/other/TabularSheet/EditField/index.js [hunk 2]
```diff
@@ -118,6 +156,23 @@ class EditFieldModal extends React.PureComponent {
...
         }
 
+        if (type === 'datetime') {
+            const selectLabel = 'Date Format';
+            const { faramValues: { options: { dateFormat } = {} } } = this.state;
+            return (
+                <SelectInput
+                    label={selectLabel}
+                    options={DATE_FORMATS}
+                    keySelector={x => x.format}
```
Comment (tnagorra) [inline src/components/other/TabularSheet/EditField/index.js]: avoid inline functions

File: src/components/other/TabularSheet/EditField/index.js [hunk 3]
```diff
@@ -118,6 +156,23 @@ class EditFieldModal extends React.PureComponent {
...
+                    labelSelector={x => x.label}
```
Comment (tnagorra) [inline src/components/other/TabularSheet/EditField/index.js]: avoid inline functions

File: src/components/other/TabularSheet/EditField/index.js (hunk 1 above)
Comment (bewakes) [reply src/components/other/TabularSheet/EditField/index.js]: fixed

File: src/components/other/TabularSheet/EditField/index.js (hunk 2 above)
Comment (bewakes) [reply src/components/other/TabularSheet/EditField/index.js]: my bad. fixed

File: src/components/other/TabularSheet/EditField/index.js (hunk 3 above)
Comment (bewakes) [reply src/components/other/TabularSheet/EditField/index.js]: my bad. fixed

**Pull request merged**